backend/cdApp.py

```diff
@@ -37,35 +37,138 @@ def dict_factory(cursor, row):
+        g.db.commit()
+        return jsonify(message='Children were successfully added to the node', id='1')
+
+    def post(self, node_id):
+        g.db.execute('''UPDATE children 
+                        SET children=(?) 
+                        WHERE parent_id=(?)''', [request.form['children'], node_id])
+        g.db.commit()
+        
+        return jsonify(message='Children were successfully updated.')
+
+
+class Tree(Resource):
+    def post(self, node_id):
+        """This rpc updates/edits a node"""
+        try:
+            g.db.execute('UPDATE nodes SET contents=(?), renderer=(?) WHERE node_id = (?)', 
+                        [request.form['contents'], request.form['renderer'],  node_id])
+
+            # Delete the old links
+            g.db.execute('DELETE FROM links WHERE origin = (?)', node_id)
+
+            # TODO(nfischer): fix this to add links (currently not working)
+            # for child in request.form['children']:
+                # print type(jsonify(child))
+                # print request.form['children']
+                # child_name = child.key()
+                # child_id = child.value()
+                # g.db.execute('INSERT INTOlinks VALUES (?, ?, ?)', [node_id, child_name, child_id])
+            g.db.commit() # Only commit if everything succeeds
+        except Exception as e:
+            print str(e)
+            # return jsonify(message='Error when editing your node')
+            raise InvalidUsage('Error when editing your node')
+        return jsonify(message='Successfully updated node %s' % node_id)
+
+    #DEPRECATED
+    # def get(self, node_id):
+    #     """This will access a node referenced by node_id"""
+
+    #     # Fetch a sqlite3.Cursor from the database
+    #     cursor = g.db.execute("SELECT * FROM nodes WHERE node_id=(?) ", node_id)
+
+    #     if cursor.rowcount > 1:
+    #         return jsonify(message='Node ID is not unique')
+
+    #     try:
+    #         ret_node = cursor.fetchall()[0]
+    #     except IndexError:
+    #         return jsonify(message='Node %s is not present in table' % node_id)
+
+    #     # Find the children of this node
+    #     cursor = g.db.execute("SELECT name, dest FROM links WHERE origin=(?) ", node_id)
+    #     children = {}
+    #     # for k, val in enumerate(cursor.fetchall()):
+
+    #     # Map each child name to an ID
+    #     for child in cursor.fetchall():
+    #         children[child['name']] = child['dest']
+    #     ret_node['children'] = children
+    #     cursor.close()
+    #     return ret_node
+
+
+    # /nodes/tree
+    # returns the tree of nodes
+    # rootId is hard coded right now. Need clarification on that
+    def get(selft):
+        cursor = g.db.execute('''SELECT n.id, n.contents, n.renderer, c.children 
+                              FROM nodes AS n LEFT JOIN children AS c 
+                              ON n.id = c.parent_id''');
+        rv = {}
```

Comment:
We should probably rename this to something more descriptive, like `node_tree`, which specifies what it represents